app/vminsert/netstorage/netstorage.go

```diff
@@ -70,7 +70,7 @@ func (sn *storageNode) push(snb *storageNodesBucket, buf []byte, rows int) error
 		// Fast path - the buffer is successfully sent to sn.
 		return nil
 	}
-	if *dropSamplesOnOverload && !sn.isReadOnly.Load() {
+	if *dropSamplesOnOverload && !sn.isBroken.Load() && sn.isReadOnly.Load() {
```

Comment:
Could you please explain which problem solves this change? 

 Previously, samples could be dropped only if storage node is overloaded and it's not read only.

With this change, samples would be dropped on overload if storage is not broken and in read only state.

Reply:
Sorry, there is a problem here, which leads to misleading.